### DISCUSSION TITLE
on mobile device im getting ClientResponseError 0

### DISCUSSION THREAD

**Discussion by Marius555** [pocketbase/pocketbase#3673] [Q&A]
im using next 14 and im trying to get AuthMethods, but im getting error ClientResponseError 0, its strage because on PC everything works ok, but on mobile device getting this error

```js
useEffect(() => {
        async function call() {
            try {
                pb.autoCancellation(false);
                const result = await pb.collection('users').listAuthMethods()
                const mod = result.authProviders !== "undefined" ? result.authProviders : null
                setupdater(mod)
            } catch (err) {
                if (!err.isAbort) {
                    alert(err)
                    console.warn("non cancellation error:", err)
                }
            }
        }
        call()
    }, [])
```

**Comment by ganigeorgiev** [comment]
`0` status code usually means a client-side error, eg. the request has failed for some reason, the connection was interrupted, invalid fetch arguments, etc.

You can inspect in more details the original thrown error by logging `err.originalError`.

If it works on desktop but fails when testing on the mobile device, then I suspect that the mobile device can't connect to your PocketBase server. You can test that by trying to open the same SDK initialization url in a browser on the mobile device.

**Reply by Marius555** [reply]
when i use "err.originalError", its say, TypeError: Failed to fetch. why its happens on mobile but not on PC?

**Reply by ganigeorgiev** [reply]
Are you sure that you can access the PocketBase server from your mobile device? Did you try opening the admin UI with the same SDK initialization url in a browser from the mobile device?

**Reply by Marius555** [reply]
i cant reach it from mobile...well difference is that pc uses localhost to react main page and 127.0.0.1 to react pocketbase, mobile device use local IP address 192.168.10.224:3000...

**Reply by ganigeorgiev** [reply]
I don't understand the above, but if the PocketBase Admin UI is not reachable when using the same url that you have in your SDK initialization, then you can:

- make sure that the mobile and desktop are on the same network and change the SDK initialization url to use the local IP of your desktop (you need to also update your local firewall to allow inbound localhost connections)

- or easier, use [ngrok](https://ngrok.com/download) to create a public address for your PocketBase server and use the generated url when initializing the SDK (aka. `new PocketBase(ngrokUrl)`).

**Reply by Marius555** [reply]
when i open local server on 192.168.10.224:3000, it renders authMethods, but when i click to auth with google, it retrives keys in localstorage, but i cant export them to cookie...

**Reply by ganigeorgiev** [reply]
If you use the cookie SDK helpers, then by default the cookie will be exported with `secure: true` attribute. It is safe to set it to `false` during development (_it works with `true` when using `localhost`/`127.0.0.1` because most browsers usually have exception for it_).

**Reply by Marius555** [reply]
do i missing something? i tried "document.cookie = pb.authStore.exportToCookie({ httpOnly: false })" but it doesnt work...

**Reply by ganigeorgiev** [reply]
If you are not connecting to localhost, try with:
```js
document.cookie = pb.authStore.exportToCookie({ httpOnly: false, secure: false })
```

**Reply by Marius555** [reply]
works fine, super...sorry im a bit new to programing...just one question...when you deploy, as i understand you have to remove both of these flags, will it work okey on deployed server without it?

**Reply by ganigeorgiev** [reply]
The `secure` flag can be removed or set to `true` once you deploy it on a host with HTTPS.
The `httpOnly` flag (_it usually has effect when the cookie is set on the server-side_) is used if you want to access the cookie both server-side and client-side.

_Side-note: I've tried to explain it in https://github.com/pocketbase/pocketbase/discussions/3574#discussioncomment-7345956, but it seems that it has caused more confusion then it actually helped, but in general my recommendation is - if you are not working on a website then don't use JavaScript SSR/RSC if you are not sure what you are getting into and instead prefer plain client-side rendering (eg. SPA) as it is a lot easier to manage and understand what is being executed and where._
